test/tests/interfacekernels/1d_interface/tests

```diff
@@ -76,5 +82,30 @@
     issues = '#13232'
     requirement = "Save-in shall record the correct residuals and Jacobians representing the interface kernel. "
                   "This applies the DirichletBC to side b (block 1) of the interface."
+    design = 'InterfaceKernels/index.md'
+  [../]
+  [./reaction_1D_steady_CSVDiff]
+    type = CSVDiff
+    input = 'reaction_1D_steady.i'
+    csvdiff = 'reaction_1D_steady_out.csv'
+    issues = '#13365'
+    requirement = "The interface reaction kernel should reproduce the analytic solution"
+    design = 'InterfaceReaction.md'
+  [../]
+  [./reaction_1D_steady_ExoDiff]
+    type = Exodiff
+    input = 'reaction_1D_steady.i'
+    exodiff = 'reaction_1D_steady_out.e'
+    issues = '#13365'
+    requirement = "The interface reaction kernel should work with interface diffusion kernel"
+    design = 'InterfaceReaction.md'
+  [../]
+  [./reaction_1D_transient]
+    type = Exodiff
+    input = 'reaction_1D_transient.i'
+    exodiff = 'reaction_1D_transient_out.e'
+    issues = '#13365'
+    requirement = "The solution should be discontinued at the interface due to the reaction taking place"
+    design = 'InterfaceReaction.md'
```

Comment:
Please also add a `PetscJacobianTest` so we can be sure that the Jacobian is good. Inspecting it, it looked perfect, but we want to make sure someone doesn't come along later and muck it up. You can see `test/tests/kernels/ad_transient_diffusion/tests` for an example

Reply:
Hi Alex, I have modified all other things except this. Take the following as an example, I do not know how to get the information for cli_args, run_sim, ratio_tol, and difference_tol. Is it somehwhere from the details in the CIVET check?

```
  [./reaction_1D_transient_Jac]
    type = PetscJacobianTester
    input = 'reaction_1D_transient.i'
    cli_args = 'Outputs/exodus=false Mesh/nx=3 Mesh/ny=3 Executioner/num_steps=3'     ##This is from test/tests/kernels/ad_transient_diffusion/tests, what should I put in my case?
    run_sim = True           ##what's this?
    ratio_tol = 1e-7         ##how could I get this?
    difference_tol = 1e-6    ##how could I get this?
    issues = '#13365'
    requirement = "The Jacobian from InterfaceReaction shall be perfect"
    design = 'InterfaceReaction.md'
  [../]
```

Reply:
What's cli_args? 
Looking at that example, for my case it should be:
    `cli_args = 'Outputs/exodus=false Mesh/nx=10 Executioner/num_steps=10'`
Why the `Outputs/exodus=false`? I have `exodus=true` in my input file.

Reply:
Hi Wesley, see my email in your thread in the MOOSE user group list to get some answers to these questions. Hope it helps!

Reply:
Since the two examples I provided (steady and transient) use the same kernel, so one Jacobian Test is enough, I will do that to the transient example. Based on your suggestion, my understanding is,  `cli_args` is the user input, and `run_sim`, `ratio_tol`, and `difference_tol` are the output we can get by running `./run_tests --dump` in the `test` (not `tests`) directory.
So, I got the information for `run_sim`, `ratio_tol`, and `difference_tol`. For `cli_args`, do you think the following would be good? Actually I have a question, in my example there would be two domains, if it is set nx=10 here, how does it know which domain has how many mesh?
  
[./reaction_1D_transient_Jac]
    type = PetscJacobianTester
    input = 'reaction_1D_transient.i'
    cli_args = 'Outputs/exodus=false Mesh/nx=10 Executioner/num_steps=3'     
    run_sim = False 
    ratio_tol = 1e-8
    difference_tol = 1e-8
    issues = '#13365'
    requirement = "The Jacobian from InterfaceReaction shall be perfect"
    design = 'InterfaceReaction.md'
  [../]

Reply:
Your mesh is already size 10, so you don't need to do anything on the command line for that.
It would be nice if you set `run_sim = True`. So this block should get the job done:
```
[./reaction_1D_transient_Jac]
type = PetscJacobianTester
input = 'reaction_1D_transient.i'
cli_args = 'Outputs/exodus=false Executioner/num_steps=3'
run_sim = True
ratio_tol = 1e-8
difference_tol = 1e-8
issues = '#13365'
requirement = "The Jacobian from InterfaceReaction shall be perfect"
design = 'InterfaceReaction.md'
[../]
```
I would honestly be surprised if your test passed with `difference_tol = 1e-8`. You may have to loosen that tolerance somewhat. That parameter is usually of secondary importance compared to `ratio_tol`. If your `ratio_tol` is good, that is usually the best indicator of a good Jacobian.

Reply:
Looks like `difference_tol` will need to be loosened. The [most recent test failure](https://civet.inl.gov/job/344326/) shows the following for time step 3 for this test (steps 1 and 2 had differences `< 1e-08`): 
```
Time Step 3, time = 0.3, dt = 0.1
 
  0 Nonlinear |R| = 3.511155e-01
    |residual|_2 of individual variables:
                           u: 0.339886
                           v: 0.0880891
 
   ---------- Testing Jacobian -------------
   ||J - Jfd||_F/||J||_F = 6.21241e-10, ||J - Jfd||_F = 7.74741e-08
       0 Linear |R| = 3.511155e-01
       1 Linear |R| = 5.157019e-16
  1 Nonlinear |R| = 1.658004e-15
     |residual|_2 of individual variables:
                          u: 1.63104e-15
                           v: 2.97807e-16
```
The ratio looks good though